### ISSUE TITLE
[Bug] when pressing <R-ALT> key, R-ALT and L-CTRL gets recognized

### ISSUE THREAD

**Issue by nnako** [Keychron/qmk_firmware#172]
### Describe the Bug

I am currently experiencing that when pressing the right ALT key (`<R-ALT>` or `KC_RALT`), not only the right ALT key but at the same time also the left CONTROL key (`<L-CTRL>` or `KC_LCTL`) gets recognized and triggered by the keyboard.

I am using a K8 PRO ISO RGB (German) keyboard and tried this using the "[usevia.app](https://usevia.app)" application on the KEY TESTER pane.

![image](https://github.com/Keychron/qmk_firmware/assets/5845602/8bee6cd7-639f-48d9-b7bb-0266f77ff877)

questions

- Does anybody know why the keyboard(s) behave like this?
- Is it an unspoken law to operate maybe only German keyboards like this?
- Is there a way to modify this low-level configuration to be able to recognize only a `KC_RALT` when a `<R-ALT>` is pressed, and no additional `KC_LCTL`?

### Keyboard Used

Keychron K8 Pro RGB (German)

### Operating System

Windows 10

### Is AutoHotKey / Karabiner installed

- [ ] AutoHotKey (Windows)
- [ ] Karabiner (macOS)

**Comment by lokher** [comment]
Can you show me the keymap in VIA?

**Comment by nnako** [comment]
Sorry. Don't understand. How do I get to the "keymap in VIA"? Is "VIA" the web application I open when issuing https://www.caniusevia.com/docs/supported_keyboards in the browser? Within their "supported keyboards" I don't see a "Keychron K8 Pro".

Could you point me to some valuable information in order to be able to answer your request?

**Comment by nnako** [comment]
any hint? any help on this?

**Comment by lokher** [comment]
Can you try holding fn + j + z  these three keys for 4 seconds to factory reset the keyboard?

**Comment by nnako** [comment]
yes, I did that. no change regarding the issue. just some configuration changed which resulted in the delay time until keyboard sleep to be lower than before. after starting "usevia.app", the delay is normal again. but the issue is still there.

**Comment by lokher** [comment]
Can you share the screenshoft as beflow of all layers with me? 
![image](https://github.com/Keychron/qmk_firmware/assets/20199239/5a61c2ee-8704-415f-ad55-72d37ff7ab46)

**Comment by nnako** [comment]
![image](https://github.com/Keychron/qmk_firmware/assets/5845602/b6f922c2-3900-454b-8614-c0e7d7e893f7)
![image](https://github.com/Keychron/qmk_firmware/assets/5845602/1b9c6ce6-8715-4523-a067-d6493cdad501)
![image](https://github.com/Keychron/qmk_firmware/assets/5845602/65e48a6e-696b-43cd-94e6-a430779a6e36)
![image](https://github.com/Keychron/qmk_firmware/assets/5845602/e571b680-290f-4c60-9835-bee87c2980d0)

the only thing I see is that on German keyboards (as mine) the `<Y>` and `<Z>` are exchanged. I use the "Windows" switch position which activates "Layer 2", as I have read.

**Comment by lokher** [comment]
Does `Left Ctrl` cause `Right Alt` pressing as well? Can you toggle on `Test Matrix` and test again to check if both keys are trigger if `Right Alt` is pressed? 
![image](https://github.com/Keychron/qmk_firmware/assets/20199239/09d6f68a-9a0d-4c91-bc0f-04330b3beff2)

**Comment by nnako** [comment]
here, I took a small video describing what you asked me to do:

https://github.com/Keychron/qmk_firmware/assets/5845602/54db5b39-8159-4318-90a1-4bba7d88d6ea

**Comment by lokher** [comment]
Can you test it on another PC? Looks like a OS issue

**Comment by nnako** [comment]
Indeed. I checked on an English Linux machine. And here, the test (without selecting "Test Matrix") shows that when pressing `<R-ALT>` only a `KC_RALT` gets detected. No `KC_LCTL`. Which is exactly what I had expected.

But what could be the reason for this? Is this a problem (or maybe it is considered as a special "feature") of only German versions of Windows OS? Would there be ways to deactivate this "feature" within the Windows OS?

**Comment by nnako** [comment]
Is there anything I can do to move forward, solving that issue locally? Any hints?

**Comment by lokher** [comment]
Can you test in safe mode of Windows? If it works fine in safe mode, try disabling unnecesary startup option.

**Comment by nnako** [comment]
are there no further hints available? where could I find more information about where to further check relevant operating system configurations?

**Comment by lokher** [comment]
> are there no further hints available? where could I find more information about where to further check relevant operating system configurations?

Have you tried safe mode?

**Comment by mbartlett21** [comment]
@nnako 
If you are on Windows, the AltGr key (which Windows maps right Alt to on some keyboard layouts) is passed to applications as Ctrl+Alt rather than a separate keycode.
